### PR TITLE
Add support for a per-view custom alt tag

### PR DESCRIPTION
**Pull request by zackdotcomputer**
**Description of changes:**
Adds support for a view to have an `alt` property that will be set as the `alt` tag on the rendered image. This will override the `caption` if present and will not render anywhere else.

This also adds a section to the README documenting the caption, alt, and source properties on ViewType.

**Related issues (if any):**
#369 - This is responding to confusion if using the HTML-as-caption feature without overriding the alt formatter.

**Checks:**

- [x] Please confirm `yarn run lint` ran successfully
- [x] Please confirm that only `/src` and `/examples/src` are committed
- [x] [if new feature] Please confirm that documentation was added to the README.md

### CODE REVIEW

**Comment by davwheat** [comment]
Looks fab! Thanks for the PR (again)!

**Comment by zackdotcomputer** [comment]
No problem at all, thanks to you to helping maintain this! I spent all day looking at react lightboxes and it's clear this one is the best free one out there, so I'd be remiss to just use it without helping out.

**Comment by zackdotcomputer** [comment]
(Fwiw I can't merge so I'll just leave this for someone with commit rights to close)

File: README.md [hunk 1]
```diff
@@ -76,3 +76,28 @@ class Component extends React.Component {
...
+
+```jsx
+const images = [{ source: 'path/to/image-1.jpg' }, { source: 'path/to/image-2.jpg' }];
+```
+
+However, react-images supports several other properties on each image object than just `source`. For example:
+
+```jsx
+const image = {
+  caption: "An image caption, or a rendered HTML string",
```
[the remark this answers is not in the document]
Comment (zackdotcomputer) [reply README.md]: Done!

File: src/formatters.js [hunk 2]
```diff
@@ -54,6 +54,8 @@ function getFullscreenLabel({ isFullscreen }: FullscreenProps): string {
 
 /* alt text for each image in the carousel */
 function getAltText({ data, index }): string {
+  if (data.alt) return data.alt;
+
   if (data.caption) return data.caption;
```
[the remark this answers is not in the document]
Comment (zackdotcomputer) [reply src/formatters.js:70]: Added both a console.error for non-string alt values, and a console.warn for a non-string caption with no alt value.

**Comment by zackdotcomputer** [comment]
Sorry @davwheat - I think my git repos were in a weird state. Force pushed this branch and it's showing up now [here](https://github.com/jossmac/react-images/pull/370/commits/b51d1b9dde245ef5e0bed9fc9edfb3693c88a82a).

**Comment by zackdotcomputer** [comment]
Oops - forgot to hit "re-request review" - lemme know if there's anything else you'd like changed here @davwheat .

**Pull request merged**